ledger/ledger-security-evidence/tool/src/scala/org/scalatest/daml/SystematicTestingGenerator.scala

```diff
@@ -0,0 +1,170 @@
+trait TestEntry[T, TS] {
+
+  /** suiteName The name of the test suite that this test belongs to. */
+  def suiteName: String
+
+  /** The description of the test. */
+  def description: String
+
+  /** The test tag to classify the test. */
+  def tag: T
+
+  /** Indicate if the test is currently ignored. */
+  def ignored: Boolean
+
+  /** Optional test suite data that applies to all tests in the test suite. */
+  def suite: Option[TS]
+}
+
+case class SecurityTestEntry(
+    suiteName: String,
+    description: String,
+    tag: SecurityTest,
+    ignored: Boolean,
+    suite: Option[SecurityTestSuite],
+) extends TestEntry[SecurityTest, SecurityTestSuite]
+
+case class ReliabilityTestEntry(
+    suiteName: String,
+    description: String,
+    tag: ReliabilityTest,
+    ignored: Boolean,
+    suite: Option[ReliabilityTestSuite],
+) extends TestEntry[ReliabilityTest, ReliabilityTestSuite]
+
+object SystematicTestingGenerator {
+
+  private def testNameWithTags(tags: Map[String, Set[String]]): List[(String, List[TestTag])] =
+    tags.fmap { tagNames =>
+      tagNames.toList
+        .filter(_.startsWith("{")) // Check if we have a JSON encoded tag
+        .traverse(decode[SystematicTesting.TestTag])
+        .valueOr(err => sys.error(s"Failed to parse JSON tag: $err"))
+    }.toList
+
+  private def isIgnored(suite: Suite, testName: String): Boolean =
+    suite.tags.getOrElse(testName, Set()).contains(Suite.IgnoreTagName)
+
+  private def scalaTestEntries[TT: ClassTag, TS: ClassTag, TE](
+      suites: List[Suite],
+      testEntry: (String, String, TT, Boolean, Option[TS]) => TE,
+  ): List[TE] = {
+    suites.flatMap { suite =>
+      val testSuite = suite match {
+        case testSuite: TS => Some(testSuite)
+        case _ => None
+      }
+
+      testNameWithTags(suite.tags).mapFilter { case (testName, testTags) =>
+        testTags.collectFirst { case testTag: TT =>
+          testEntry(suite.suiteName, testName, testTag, isIgnored(suite, testName), testSuite)
+        }
+      }
+    }
+  }
+
+  private def testEntries[TT: ClassTag, TS: ClassTag, TE](
+      suites: List[LedgerTestSuite],
+      testEntry: (String, String, TT, Boolean, Option[TS]) => TE,
+  ): List[TE] = {
+    suites.flatMap { suite =>
+      val testSuite = suite match {
+        case testSuite: TS => Some(testSuite)
+        case _ => None
+      }
+
+      val tags = suite.tests.map { test =>
+        test.name -> test.tags.map(_.name).toSet
+      }.toMap
+
+      testNameWithTags(tags).mapFilter { case (testName, testTags) =>
+        testTags.collectFirst { case testTag: TT =>
+          testEntry(suite.name, testName, testTag, false, testSuite)
+        }
+      }
+    }
+  }
+
+  private def loadIntelliJClasspath(): Option[String] =
+    Some(System.getProperty("java.class.path")).filter(!_.contains("sbt-launch.jar"))
+
+  def main(args: Array[String]): Unit = {
+
+    val ledgerApiTests = (suites.v1_8.default(0L) ++ suites.v1_8.optional()).toList
```

Comment:
Not ideal that we need to depend on the contents of `//ledger/ledger-api-tests/suites`. Can we do without? (e.g. see my other comment re annotations)

Reply:
How would that fix the situation? We would still need an access to these on the classpath level